tsl/test/expected/continuous_aggs_drop_chunks.out

```diff
@@ -139,3 +139,136 @@ SELECT drop_chunks('records', '2000-03-16'::timestamptz);
+    $$ SELECT 125 $$;
+CREATE TABLE conditions(time_int INT NOT NULL, device INT, value FLOAT);
+SELECT create_hypertable('conditions', 'time_int', chunk_time_interval => 10);
+    create_hypertable    
+-------------------------
+ (3,public,conditions,t)
+(1 row)
+
+INSERT INTO conditions
+SELECT time_val, time_val % 4, 3.14 FROM generate_series(0,100,1) AS time_val;
+SELECT set_integer_now_func('conditions', 'test_int_now');
+ set_integer_now_func 
+----------------------
+ 
+(1 row)
+
+CREATE MATERIALIZED VIEW conditions_7
+    WITH (timescaledb.continuous, timescaledb.materialized_only = TRUE)
+    AS
+        SELECT time_bucket(7, time_int) as bucket,
+            SUM(value), COUNT(value)
+        FROM conditions GROUP BY bucket WITH DATA;
+NOTICE:  refreshing continuous aggregate "conditions_7"
+HINT:  Use WITH NO DATA if you do not want to refresh the continuous aggregate on creation.
+SELECT materialization_hypertable_schema||'.'||materialization_hypertable_name AS mat_hyper
+FROM timescaledb_information.continuous_aggregates
+WHERE view_name = 'conditions_7' \gset
+CREATE VIEW see_cagg AS SELECT * FROM conditions_7 WHERE bucket < 70 ORDER BY bucket;
+--CREATE VIEW see_mat AS SELECT bucket, count(*) FROM :mat_hyper WHERE bucket < 70 
+--GROUP BY bucket ORDER BY bucket;
+SELECT * FROM see_cagg;
+ bucket |  sum  | count 
+--------+-------+-------
+      0 | 21.98 |     7
+      7 | 21.98 |     7
+     14 | 21.98 |     7
+     21 | 21.98 |     7
+     28 | 21.98 |     7
+     35 | 21.98 |     7
+     42 | 21.98 |     7
+     49 | 21.98 |     7
+     56 | 21.98 |     7
+     63 | 21.98 |     7
+(10 rows)
+
+-- This is the simplest case, when the updated bucket is inside chunk, so it is expected
+-- always reflected.
+UPDATE conditions SET value = 4.00 WHERE time_int = 2;
+SELECT drop_chunks('conditions', 20);
+               drop_chunks               
+-----------------------------------------
+ _timescaledb_internal._hyper_3_65_chunk
+ _timescaledb_internal._hyper_3_66_chunk
+(2 rows)
+
+SELECT * FROM see_cagg;
+ bucket |  sum  | count 
+--------+-------+-------
+      0 | 22.84 |     7
+      7 | 21.98 |     7
+     14 | 21.98 |     7
+     21 | 21.98 |     7
+     28 | 21.98 |     7
+     35 | 21.98 |     7
+     42 | 21.98 |     7
+     49 | 21.98 |     7
+     56 | 21.98 |     7
+     63 | 21.98 |     7
+(10 rows)
+
+-- This case is update in a bucket, which crosses two chunks, which exist and both is going
+-- to be dropped.
+UPDATE conditions SET value = 4.00 WHERE time_int = 31;
+-- This update is in the bucket, which crosses two chunks. The first chunk is going to be 
+-- dropped now and the update is outside it, and thus should not be reflected. The second 
+-- chunk contains the update and will be dropped on the next call, thus it should be reflected 
+-- in the continuous aggregate only after the last drop.
+UPDATE conditions SET value = 4.00 WHERE time_int = 41;
```

Comment:
Isn't a more interesting test case one where you update the _same_ bucket twice, but each update falls within a separate partial. Then when you drop the first chunk, the first partial of bucket X will be updated, but not the second partial of bucket X. For instance,

```
UPDATE conditions SET value = 4.00 WHERE time_int = 29;
UPDATE conditions SET value = 4.00 WHERE time_int = 32;
```

These two updates fall in the same bucket `[28,35)`, but are two different partials since the chunk border is at 30.

Reply:
It is not more interested than the current tests. It doesn't allow to see if the fix recognises, which partials need to be updated, and this test was already working correctly before this fix.
I will add it.

Reply:
Maybe I am missing something, but if a bucket can be split across two partials (one in each chunk), it seems reasonable to test updates in those two same-bucket partials to see that both cases are properly reflected in the bucket when dropping the corresponding chunks.

For instance, bucket 2 below is split across two chunks, thus consisting of two partials 2a and 2b. When dropping the first chunk we'd like to see that bucket 2 is updated to reflect changes in chunk 1 and when the second chunk is dropped we'd like to see that bucket 2 is updated to reflect changes in chunk 2.

```
  1    2     3
|----|-----|----| buckets
       a|b
|-------|-------| chunks
```

Reply:
The described case is already producing correct result before this fix because it was refreshing bucket 2 two times, so the test doesn't demonstrated the corner case described by the issue.
This PR fixes that 2b is refreshed during drop of chunk b, and 2a is refreshed during drop of chunk a.  The regression test was already testing the most interesting cases. I added the test as you requested.